helix-term/src/ui/statusline.rs

```diff
@@ -451,13 +451,15 @@ fn render_file_modification_indicator<F>(context: &mut RenderContext, write: F)
 where
     F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
 {
-    let title = (if context.doc.is_modified() {
-        "[+]"
-    } else {
-        "   "
-    })
+    let is_modified = context.doc.is_modified();
+    let externally_modified = context.doc.externally_modified;
+    let title = match (is_modified, externally_modified) {
+        (true, true) => "[!]",
+        (true, false) => "[+]",
+        (false, true) => "[-]",
```

Comment:
```suggestion
        (false, true) => "[~]",
```
`-` does not have connotations of "this file has been externally modified". `~` may be a better suited symbol for this

Reply:
I'm not particularly opinionated about the symbol here, but let me explain the reasoning for `-` and see what you think:

`-` is the logical opposite of `+`.  `+` is already used to indicate that the current state is *ahead* of the base file.  `-` makes sense for the opposite, i.e. the editor state is *behind* the base file.  Of course this is only in terms of timestamps, none of these actually concern the content (i.e. the existing `+` is already used in the case where, say, you remove some lines and haven't saved yet).

Still prefer `~`?